### PR TITLE
Feat PUD-1061 (pci.storage.databases) - Fix on availabilities/capabilities for Manager display

### PR DESCRIPTION
**Pull request by jperchoc**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/PUD-833` for bug fixes 
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #PUD-1061
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Reflect backend changes on the storage: 
- do not use the storage property on falvors in /capabilities
- use the minDiskSize and maxDiskSize properties in /availabilities instead
- if min and max disk size are the same, only show one value
- If min and max disk size are different, then show a range

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
   "pci_databases_general_information_delete_nodes": "Supprimer un nœud",
   "pci_databases_general_information_node_type_label": "Modèle de nœud",
   "pci_databases_general_information_node_type": "{{coreNumber}} CPU / {{memorySize}} GB RAM / {{storageSize}} GB Stockage SSD",
+  "pci_databases_general_information_node_type_disk_range": "{{coreNumber}} CPU / {{memorySize}} GB RAM / de {{storageMinSize}} à {{storageMaxSize}} GB Stockage SSD",
```
[the remark this answers is not in the document]
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/translations/Messages_fr_FR.json:11]: i suggest to you to pass value + Unit as parameter like:

```suggestion
  "pci_databases_general_information_node_type_disk_range": "{{coreNumber}} CPU / {{memorySize}} RAM / de {{storageMinSize}} à {{storageMaxSize}} Stockage SSD",
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/general-information.html [hunk 2]
```diff
@@ -202,11 +202,22 @@
...
+                            }"
+                            ></span>
+                            <span
+                                ng-if="$ctrl.getCurrentFlavor().minDiskSize !== $ctrl.getCurrentFlavor().maxDiskSize"
+                                data-translate="pci_databases_general_information_node_type_disk_range"
+                                data-translate-values="{
+                                coreNumber: $ctrl.getCurrentFlavor().core,
+                                memorySize: $ctrl.getCurrentFlavor().memory,
+                                storageMinSize: $ctrl.getCurrentFlavor().minDiskSize,
+                                storageMaxSize: $ctrl.getCurrentFlavor().maxDiskSize,
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/general-information.html:220]: Related to this comment : 

```suggestion
                                storageMinSize: $ctrl.getCurrentFlavor().minDiskSize | cucBytes:2:false:'GB', // GB if API provide base unit value as GB, if default base is B, so use B and so on
                                storageMaxSize: $ctrl.getCurrentFlavor().maxDiskSize | cucBytes:2:false:'GB',
```

File: packages/manager/modules/pci/src/components/project/storages/databases/flavor.class.js [hunk 3]
```diff
@@ -18,6 +17,14 @@ export default class Flavor {
     return some(this.availability, { network: networkName });
   }
 
+  get minDiskSize() {
+    return get(head(this.availability), 'minDiskSize');
+  }
+
+  get maxDiskSize() {
+    return get(head(this.availability), 'maxDiskSize');
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/components/project/storages/databases/flavor.class.js:25]: i think we can avoid to use Lodash get and head functions.

```suggestion
    return (this.availability || [])[0]?.maxDiskSize;
```

File: packages/manager/modules/pci/src/components/project/storages/databases/flavor.class.js [hunk 4]
```diff
@@ -18,6 +17,14 @@ export default class Flavor {
...
+  }
+
   get nodesCount() {
     return get(head(this.availability), 'minNodeNumber');
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/components/project/storages/databases/flavor.class.js:29]:
```suggestion
    return (this.availability || [])[0]?.minNodeNumber;
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/translations/Messages_fr_FR.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/translations/Messages_fr_FR.json]: Good suggestion, thank you ! I have made the changes here: https://github.com/ovh/manager/pull/5932/commits/ee7b1386639352dac98ee930dcb1e2b39891123c

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/general-information.html (hunk 2 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/general-information.html]: Done here : https://github.com/ovh/manager/pull/5932/commits/ee7b1386639352dac98ee930dcb1e2b39891123c

File: packages/manager/modules/pci/src/components/project/storages/databases/flavor.class.js (hunk 3 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/components/project/storages/databases/flavor.class.js]: Done here: https://github.com/ovh/manager/pull/5932/commits/ee7b1386639352dac98ee930dcb1e2b39891123c

File: packages/manager/modules/pci/src/components/project/storages/databases/flavor.class.js (hunk 4 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/components/project/storages/databases/flavor.class.js]: Done here: https://github.com/ovh/manager/pull/5932/commits/ee7b1386639352dac98ee930dcb1e2b39891123c

**Review by mohammed-zahaf — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**